pkg/crane/options.go

```diff
@@ -47,26 +52,44 @@ func makeOptions(opts ...Option) Options {
 		},
 		Keychain: authn.DefaultKeychain,
 	}
+
 	for _, o := range opts {
 		o(&opt)
 	}
+
+	// Allow for untrusted certificates if the user passed Insecure but no custom
+	// transport.
+	if opt.insecure && !opt.transportSet {
+		opt.transport = remote.DefaultTransport.(*http.Transport).Clone()
+		opt.transport.TLSClientConfig = &tls.Config{
+			InsecureSkipVerify: true, //nolint: gosec
+		}
+
+		WithTransport(opt.transport)(&opt)
+	}
+
 	return opt
 }
 
 // Option is a functional option for crane.
 type Option func(*Options)
 
 // WithTransport is a functional option for overriding the default transport
-// for remote operations.
+// for remote operations. Setting a transport will override the Insecure option's
+// configuration allowing for image registries to use untrusted certificates.
 func WithTransport(t http.RoundTripper) Option {
 	return func(o *Options) {
 		o.Remote = append(o.Remote, remote.WithTransport(t))
+		o.transportSet = true
```

Comment:
What do you think about `o.transport = t` instead of the bool, then have `makeOptions` check for `opt.transport == nil` rather than having both the bool and the transport?

Reply:
I'm thinking we'd still want to track the insecure bool, but otherwise that could work and removes a bit of a redundancy. I'll put it together and send it here momentarily (along with the Boilerplate check fix).

Reply:
Implemented in https://github.com/google/go-containerregistry/commit/5112fe351dcc371dfe427daf1870978a4c08053e